ext/uri/php_uri.h

```diff
@@ -27,7 +27,7 @@ typedef struct php_uri {
 	zend_string *user;
 	zend_string *password;
 	zend_string *host;
-	unsigned short port;
+	zend_long port;
```

Comment:
Wouldn't using an unsigned type be better?
```suggestion
	zend_ulong port;
```

Reply:
Yesn't. Technically yes, they can't be negative. But port numbers exceeding `zend_long` are already rejected during parsing, as userland can't represent them. I've opted for `zend_long` for internal consistency.

It's also getting interesting for the “consumers” here, they likely don't expect port numbers exceeding 65535 either. But they are technically valid (just not for TCP/UDP).

Reply:
ACK, maybe add a comment about this to explain the intent?